docs/developer-guide/project-structure.md

```diff
@@ -1,59 +1,118 @@
 # Project structure
-
 ## Modules
-
 ### Core Modules
 
-All the core modules are located under the [Strongbox] project. 
-Each of these modules has a `README.md` file explaining briefly what kind of code contains. 
-New modules should also follow the same format.
+All the core modules are located under the [Strongbox] project. Each of these modules has a `README.md` file explaining briefly what kind of code contains. New modules should also follow the same format.
 
 This is a brief breakdown of the modules:
 
-* [strongbox-client]/  
+* [strongbox-aql](https://github.com/strongbox/strongbox/tree/master/strongbox-aql)
+    This is where the [Artifact Query Language (AQL)](https://strongbox.github.io/user-guide/artifact-query-language.html) is located.
+    
+* [strongbox-client]    
     This is where the artifact client resides.
-* [strongbox-commons]/  
-    This is where the most common code which is across other modules resides.
-* [strongbox-event-api]/    
-    Contains an event API.
-* [strongbox-metadata-core]/  
-    This contains the most commonly needed code related to Maven metadata.
-* [strongbox-parent]  
-    This is the Maven parent which is (and should be) inherited by all modules. It is the right (and only place) 
-    to define versions for dependencies and plugins. Versions of dependencies should not be defined in any other 
-    `pom.xml` files in order to ease the maintenance of these across the codebase.
-* [strongbox-resources]/  
-    This is the place where common resources which can be used by multiple projects reside. The idea is not have to duplicate 
-    things such as `logback.xml`, `web.xml`, keystores and so on across the other modules. These resources are 
-    copied using the `maven-dependency-plugin`.
-    * [strongbox-common-resources]/  
-        Contains `logback.xml` and keystores.
-    * [strongbox-storage-resources]/  
-        * [strongbox-storage-api-resources]/  
-            Contains the `strongbox.xml` configuration file.
-        * [strongbox-web-resources]/  
-            Contains the `web.xml`and the Jetty configuration files.
-* [strongbox-rest-client]/  
+
+* [strongbox-commons] 
+    This is where the most common code that is shared across modules resides.
+
+* [strongbox-configuration]
+    Contains configuration parsing related code.
+
+* [strongbox-cron]
+
+    * [strongbox-cron-api]
+        Contains the Cron API code, that is required by custom cron and controller implementations.
+
+    * [strongbox-cron-tasks]
+        Contains the common built-in cron tasks.
+
+* [strongbox-data-service]
+    Contains the base implementation of data service classes. Check the [Getting Started With Persistence](https://strongbox.github.io/developer-guide/getting-started-with-persistence.html) article for more details.
+
+* [strongbox-distribution]
```

Comment:
There is a missing link here. Also, I think we can safely move the text from here:

```suggestion
* [strongbox-distribution]  
    This module produces the final binary distributions which are then made public under the [releases](https://github.com/strongbox/strongbox/releases) section.
```

Note: I'm using a link here, because `releases` is a common word and if you set a reference link this will make all `releases` words in the text as links.